### ISSUE TITLE
Authenticate by Library ID/PIN by FOLIO

### ISSUE THREAD

**Issue by cbeer** [sul-dlss/mylibrary#740]
(Requires mod-users 18.3)

Authenticating by PIN seems like it takes two queries, one to get the patron's uuid and one to validate the PIN:

`GET /users?query=barcode=="barcode"` (if we didn't escape the sunetid, we definitely need to escape the user-supplied barcode)
`POST /patron-pin/verify` with a json body of e.g. `{ id: uuid, pin: params[:pin] }`

`verify` will return a 200 on success, or some error code if the pin does not match.

Potential TODO: verify that FOLIO validates PINs in a constant time.

**Comment by thatbudakguy** [comment]
see https://github.com/sul-dlss/mylibrary/issues/788#issuecomment-1518350581

provided we find a way to either migrate everyone's PINs to FOLIO passwords or cause them to be set some other way, you can do this with:
```rb
  def login(library_id, pin)
    response = get_json('/bl-users/login', json: { username: library_id, password: pin }, method: :post)
    response['user']
  end
```

...because the `/bl-users/login` endpoint returns both a token and the user info on a success.

**Comment by cbeer** [comment]
Fixed by #813
